cmd/otel-allocator/main.go

```diff
@@ -126,11 +145,7 @@ type server struct {
 	server           *http.Server
 }
 
-func newServer(log logr.Logger, allocator *allocation.Allocator, discoveryManager *lbdiscovery.Manager, cliConf config.CLIConfig) (*server, error) {
-	k8sclient, err := configureFileDiscovery(log, allocator, discoveryManager, context.Background(), cliConf)
-	if err != nil {
-		return nil, err
-	}
+func newServer(log logr.Logger, allocator *allocation.Allocator, discoveryManager *lbdiscovery.Manager, k8sclient *collector.Client, listenAddr *string) (*server, error) {
```

Comment:
Not clear why this change is being made.  It seems unrelated to the allocation strategy change.

Reply:
This needs to be made in order to load the config which contains the desired allocation strategy at the start of main, rather than in a method called in newServer. Prior to the instantiation of the allocator, we need an instance of the strategy.